locations/spiders/piggly_wiggly_us.py

```diff
@@ -1,11 +1,25 @@
-from locations.storefinders.freshop import FreshopSpider
+import json
+import re
 
+from scrapy.spiders import SitemapSpider
 
-class PigglyWigglyUSSpider(FreshopSpider):
+from locations.dict_parser import DictParser
+
+
+class PigglyWigglyUSSpider(SitemapSpider):
     name = "piggly_wiggly_us"
     item_attributes = {"brand": "Piggly Wiggly", "brand_wikidata": "Q3388303"}
-    app_key = "piggly_wiggly_nc"
+    sitemap_urls = ["https://www.pigglywiggly.com/wp-sitemap-posts-page-1.xml"]
+    sitemap_rules = [(r"/store-locations/.+/", "parse")]
+
+    def parse(self, response, **kwargs):
+        locations = json.loads(re.search(r"locations = (\[?{.+}\]?);", response.text).group(1))
```

Comment:
`.+` would be greedy and return beyond the end of that dictionary, possibly into later dictionaries in the same JavaScript blob?

Would the following work instead:
```
from chompjs import parse_js_object

locations = parse_js_object(response.xpath('//script[contains(text(), "var locations = ")]/text()').get().split("var locations = ", 1)[1].split("\n", 1)[0])
```

Reply:
The current regex wont go beyond the line, and if they minimised it in future it would break on the first bit

```patch
-locations = (
+locations=(
```

I try to avoid chompjs as much as possible now, it's broken too many times, it's our only option for javascript. But for json, I skip it as much as possible.

Reply:
I think the issue would be if a string in the JSON blob contained `];` but as you said, this would cause the JSON blob to stop being parsed half way and we'd see the error as 0 or a smaller number of locations returned for this spider than usual.